designs/oauth.rst

```diff
@@ -93,21 +93,21 @@ made based on other factors like group membership.
 
 ::
 
-    class: oauth_resource_server
-    class: oauth_resource_server_basic
-    oauth_rs_name: String,
-    oauth_rs_basic_secret: String,
+    class: oauth2_resource_server
+    class: oauth2_resource_server_basic
+    oauth2_rs_name: String,
+    oauth2_rs_basic_secret: String,
     # To validate the redirect root
-    oauth_rs_origin: String/URI
+    oauth2_rs_origin: String/URI
     # Scopes that apply to all users
-    oauth_rs_scope_implicit: String
+    oauth2_rs_scope_implicit: String
     # Scopes that map to groups which will be enforced.
-    oauth_rs_scope_map: (String, reference)
+    oauth2_rs_scope_map: (String, reference)
     # Filter of accounts that may authorise through this.
-    oauth_rs_account_filter: Filter
+    oauth2_rs_account_filter: Filter
     # A per-resource server fernet key for token/codes.
     # Allows reset per/application in case of suspect compromise.
-    oauth_rs_token_key: String
+    oauth2_rs_token_key: String
 
 The returned authorisation code should be fernet encrypted and contains the unsigned UAT content of the authorised
 user.
```

Comment:
What is fernet encrypted and why do we need it?

Reply:
Also, what does "Claims will be mapped to a kanidm namespace" mean?

Reply:
Fernet ( https://github.com/fernet/spec/blob/master/Spec.md ) is a way of encrypting a data blob that is unable to be inspected OR altered by an external party. It's created by cryptographers in a way to create something that "can not be held incorrectly". 

In this the reason we need a fernet key for our resource server, is so that:

* When we return the AuthorisePermitSuccess ( kanidmd/src/lib/core/https/oauth2.rs line ~170 ) that the resource server is unable to tamper with or extract the content of code for the subsequent exchange.
* That when the resource server exchanges the code for an authorisation token, the authorisation token is encrypted and can not be tampered with.

Above and beyond this, the reason each resource server needs a unique fernet key, is so that an authorisation code from one resource server can not be used by another. It also means generated tokens can not be "shifted" between resource servers either. Finally, if a resource server is compromised, we can regenerate the fernet keys of the single affected resource server to force a re-issue of any inflight code exchanges or tokens, without disrupting other resource servers that are attached to kanidm.


The meaning of claims being mapped, means simply that if we issue claims into a token response or oidc response, we will prefix this with "kanidm_<CLAIM NAME>". For example, there are will known "claim" names such as sub, iss, exp ( https://tools.ietf.org/html/rfc7662 ) , and in order to prevent conflicts with these, we have a namespacing prefix to show clearly we are the creator of that value. Consider claims to be "properties" about the identity that we issue into their auth token, or token introspection calls, that can be used to make extended decisisons about the identity by the resource server.